types/kagekiri.d.ts

```diff
@@ -10,4 +10,4 @@ export function querySelectorAll(selector: string, context?: Node): Element[];
 export function getElementsByClassName(names: string, context?: Node): Element[];
 export function getElementsByTagName(tagName: string, context?: Node): Element[];
 export function getElementsByTagNameNS(namespaceURI: string, localName: string, context?: Node): Element[];
-export function getElementById(id: string): Element | null;
+export function getElementById(id: string, context?: DocumentOrShadowRoot): Element | null;
```

Comment:
Maybe we should make this `Document | ShadowRoot | HTMLDcoument`?

https://github.com/microsoft/TypeScript/blob/5305e4a44e4e08c911d181e207602bd4cf335ba1/src/lib/dom.generated.d.ts#L4960-L4977

I don't trust `DocumentOrShadowRoot` maybe because it's a Chrome-only mixin

Reply:
Is it Chrome-only? I got it from [MDN](https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot), also it looks like it's in [the HTML spec](https://dom.spec.whatwg.org/#mixin-documentorshadowroot)

Reply:
I saw [this MDN article](https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot) which reads:

> The ShadowRoot interface includes the following methods defined on the DocumentOrShadowRoot mixin. Note that this is currently only implemented by Chrome; other browsers still implement them on the Document interface.

But I must be misunderstanding because the [MDN for DocumentOrShadowRoot](https://developer.mozilla.org/en-US/docs/Web/API/DocumentOrShadowRoot) displays support in all browsers.

Reply:
Hmmm yeah I'm not sure what they mean by that. There is no `DocumentOrShadowRoot` global even defined in Chrome, and the mixin's methods look like generic stuff that would apply to either Document or ShadowRoot (e.g. `activeElement`). In any case since this is for TypeScript, yeah I think we're fine just following the spec